frontend/src/components/pages/CampOverview/ManageSessions/ManageSessionsModal.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState } from "react";
+import {
+  Button,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from "@chakra-ui/react";
+import { ManageCampSessionDetails } from "../../../../types/CampsTypes";
+import ManageSessionsModalTable from "./ManageSessionsModalTable";
+
+interface ManageSessionsModalProps {
+  campStartTime: string;
+  campEndTime: string;
+  sessions: Array<ManageCampSessionDetails>;
+  onSaveChanges: (
+    deletedSessions: Set<string>,
+    updatedCapacities: Map<string, number | undefined>,
+  ) => void;
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const ManageSessionsModal = ({
+  campStartTime,
+  campEndTime,
+  sessions,
+  onSaveChanges,
+  isOpen,
+  onClose,
+}: ManageSessionsModalProps): JSX.Element => {
+  const [deletedCampSessionIds, setDeletedCampSessionIds] = useState<
+    Set<string>
+  >(new Set());
+  const [updatedSessionIdCapacity, setUpdatedSessionIdCapacity] = useState<
+    Map<string, number | undefined>
```

Comment:
Can we not avoid the undefined here? If the user removes the value in the field, can you remove session from this map or just populate with its existing value? Unless I missed something, I think it would be cleaner to avoid the undefined and keep this state purely the session and updated capacity

Reply:
^ this was the change